nautobot/dcim/tests/test_models.py

```diff
@@ -1264,6 +1264,27 @@ def test_child_devices_are_not_saved_when_unnecessary(self):
         self.assertNotEqual(child_mtime_after_parent_rack_update_save, child_mtime_after_parent_site_update_save)
 
 
+class DeviceBayTestCase(ModelTestCases.BaseModelTestCase):
+    model = DeviceBay
+
+    def setUp(self):
+        devices = Device.objects.all()[:3]
+        devices[1].device_type.subdevice_role = SubdeviceRoleChoices.ROLE_CHILD
+        devices[1].device_type.validated_save()
+        DeviceBay.objects.create(device=devices[0], name="Device Bay 1", installed_device=devices[1])
+
+    def test_assigning_installed_device(self):
+        chassis = Device.objects.first()
+        server = Device.objects.exclude(device_type__subdevice_role=SubdeviceRoleChoices.ROLE_CHILD).last()
+        bay = DeviceBay(device=chassis, name="Device Bay Err", installed_device=server)
+        with self.assertRaises(ValidationError) as err:
+            bay.validated_save()
+            self.assertEqual(
+                err.msg["installed_device"][0],
+                "Cannot install the specified device; device subdevice role is not a child.",
+            )
```

Comment:
I don't think the assert will be reached here since the validated_save will raise an exception. Perhaps:

```suggestion
        with self.assertRaises(ValidationError) as err:
            bay.validated_save()
        self.assertEqual(
            err.msg["installed_device"][0],
            "Cannot install the specified device; device subdevice role is not a child.",
        )
```